app/models/product.rb

```diff
@@ -0,0 +1,27 @@
+class Product < ActiveRecord::Base
+  belongs_to :robot
+  has_and_belongs_to_many :categories
+  has_many :reviews
+  has_many :orderitems
+
+  validates :name, presence: true
+  validates :robot_id, presence: true
+  validates :price, presence: true, numericality: { greater_than: 0 }
+  validates :stock, numericality: {greater_than_or_equal_to: 0}
+
+  def rating_average
+    sums = []
+    self.reviews.each do |review|
+      sums.push(review.rating)
+    end
+    return sums.inject(:+) / self.reviews.count
```

Comment:
Interesting use of the `push` and `inject` here! It would probably be less processing time if you summed up the rating values rather than pushing to an array and then using an `inject`.